cohortextractor/query_engines/mssql.py

```diff
@@ -188,7 +188,7 @@ def list_query_nodes_from_category_definitions(self, definitions):
         # here because query nodes are Values with overloaded lt/gt operators.
         return sorted(
             self.get_query_nodes_from_category_definitions(definitions),
-            key=lambda x: (x.column, x.source),
+            key=lambda x: (x.column, repr(x.source)),
```

Comment:
This fixes a bug that came up in the previous_covid categorisation for the real study.  If x.source is a `Row`, it can't be used as a key in the sort.  This should be a temporary fix until we change the types of the Value/Comparator things, and/or implement proper hashes

Reply:
I think that deserves a code comment rather that just a comment on the PR. I certainly would be surprised by that code if I came to it in a few weeks.

Reply:
Good point, will add one